app/lib/backend_api_logic/routing_policy.rb

```diff
@@ -10,10 +10,17 @@ def policy_chain
       other_routing_rules = other_routing_policies.flat_map { |policy| policy['configuration']['rules'] }
 
       routing_policy = Builder.new(service).to_h
-
+      
       return other_policies if routing_policy['configuration']['rules'].concat(other_routing_rules).empty?
 
-      [routing_policy, *other_policies].compact
+      index_apicast = other_policies.index { |policy| policy['name'] == 'apicast'}
+      result = []
+      if index_apicast > 0
+        result.push(*other_policies[0..index_apicast - 1])
+      end
+      result << routing_policy
+      result.push(*other_policies[index_apicast..-1])
+      return result
```

Comment:
You should try https://ruby-doc.org/core-2.5.0/Array.html#method-i-insert here.

Reply:
Just one raw idea here:

```ruby
apicast_policy_index = other_policies.index { |policy| policy['name'] == 'apicast'}
other_policies.insert(apicast_policy_index || 0, routing_policy).compact
```

Reply:
@guicassolato This looks good, I will update it. Just a question on `apicast_policy_index || 0` , I assume this to handle the case when there is no apicast policy, but AFAIU 3scale Apicast policy is default and will always be there and cannot be deleted. Is that the case  or not ?

Reply:
You are right. apicast policy should always be there and there's no need for the `|| 0` fallback.